### PR TITLE
feat: Added PyMuPDF loader for PDF files

### PR DESCRIPTION
**Pull request by iamarunbrahma**
This PDF loader is the fastest among all other PDF parsing options and text extraction from PDF files. The logic of this pdf loader is to extract the text and also extract basic metadata (if needed) such as page number, total number of pages in pdf file and also file path.

### CODE REVIEW

File: loader_hub/file/pymu_pdf/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# PyMuPDF Loader
+
+This loader extracts text from a local PDF file using the `PyMuPDF` Python library. This is the fastest among all other PDF parsing options available in `loader_hub`. If `metadata` is passed as True during initialization; extracted documents will include basic metadata such as page numbers, file path and total number of pages in pdf.
```
Comment (jerryjliu) [inline loader_hub/file/pymu_pdf/README.md:3]: is it actually the "fastest"?

**Review by jerryjliu — APPROVED** [review]

**Comment by jerryjliu** [comment]
there's some test failures unrelated to this PR, we can help fix

File: loader_hub/file/pymu_pdf/README.md (hunk 1 above)
Comment (iamarunbrahma) [reply loader_hub/file/pymu_pdf/README.md:3]: as mentioned here (https://medium.com/social-impact-analytics/comparing-4-methods-for-pdf-text-extraction-in-python-fd34531034f). I have also personally used it and it seemed faster compared to PyPDF and PDFMiner.

**Comment by iamarunbrahma** [comment]
> there's some test failures unrelated to this PR, we can help fix

@jerryjliu  - Thanks, please let me know if any modifications need to be done from my end.

File: loader_hub/file/pymu_pdf/base.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+
+class PyMuPDFReader(BaseReader):
+    """Read PDF files using PyMuPDF library."""
+
+    def __init__(self, file_path: Union[Path, str], metadata: bool = True) -> None:
+        super().__init__(file_path)
```
Comment (painor) [inline loader_hub/file/pymu_pdf/base.py:19]: I don't think BaseReader expects file path here unless I'm using an older version

File: loader_hub/file/pymu_pdf/base.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            metadata_dict["total_pages"] = len(doc)
+            metadata_dict["file_path"] = self.file_path
+            if not extra_info:
+                extra_info = metadata_dict
+            else:
+                extra_info = dict(extra_info, **metadata_dict)
+
+            return [
+                Document(
+                    page_content=page.get_text().encode("utf-8"),
```
Comment (painor) [inline loader_hub/file/pymu_pdf/base.py:59]: same here it's "text" for me not page_content

**Pull request merged**